packages/grid/_modules_/grid/GridComponent.tsx

```diff
@@ -94,70 +95,66 @@ export const GridComponent = React.forwardRef<HTMLDivElement, GridComponentProps
-                    ContentComponent={components.ColumnMenu}
-                    contentComponentProps={{
-                      ...componentBaseProps,
-                      ...props.componentsProps?.columnMenu,
-                    }}
+                  <components.ErrorOverlay
+                    {...errorProps}
+                    {...componentBaseProps}
+                    {...props.componentsProps?.errorOverlay}
                   />
-                  <Watermark licenseStatus={props.licenseStatus} />
-                  <GridColumnsContainer ref={columnsContainerRef}>
-                    <ColumnsHeader ref={columnsHeaderRef} />
-                  </GridColumnsContainer>
-                  {showNoRowsOverlay && (
-                    <components.NoRowsOverlay
-                      {...componentBaseProps}
-                      {...props.componentsProps?.noRowsOverlay}
-                    />
-                  )}
-                  {props.loading && (
-                    <components.LoadingOverlay
-                      {...componentBaseProps}
-                      {...props.componentsProps?.loadingOverlay}
-                    />
-                  )}
-                  <GridWindow ref={windowRef}>
-                    <Viewport ref={renderingZoneRef} />
-                  </GridWindow>
                 </GridMainContainer>
-                {!gridState.options.hideFooter && (
-                  <div ref={footerRef}>
-                    <components.Footer {...componentBaseProps} {...props.componentsProps?.footer} />
-                  </div>
+              )}
+            >
+              <div ref={headerRef}>
+                <components.Header {...componentBaseProps} {...props.componentsProps?.header} />
+              </div>
+              <GridMainContainer>
+                <GridColumnHeaderMenu
+                  ContentComponent={components.ColumnMenu}
+                  contentComponentProps={{
+                    ...componentBaseProps,
+                    ...props.componentsProps?.columnMenu,
+                  }}
+                />
+                <Watermark licenseStatus={props.licenseStatus} />
+                <GridColumnsContainer ref={columnsContainerRef}>
+                  <ColumnsHeader ref={columnsHeaderRef} />
+                </GridColumnsContainer>
+                {showNoRowsOverlay && (
+                  <components.NoRowsOverlay
+                    {...componentBaseProps}
+                    {...props.componentsProps?.noRowsOverlay}
+                  />
+                )}
+                {props.loading && (
+                  <components.LoadingOverlay
+                    {...componentBaseProps}
+                    {...props.componentsProps?.loadingOverlay}
+                  />
                 )}
-              </ErrorBoundary>
-            </GridRoot>
-          )}
-        </AutoSizer>
+                <AutoSizer onResize={onResize} nonce={props.nonce} disableHeight={props.autoHeight}>
```

Comment:
Can we abstract NoSSR in AutoSizer and can we put AutoSizer in Viewport. It does not need to be visible in the root component

Reply:
> It does not need to be visible in the root component

We can abstract it, why not

> abstract NoSSR in AutoSizer

I don't understand, what do you mean?

Reply:
We can move the `NoSsr` into the `AutoSizer` but then the `AutoSizer` needs to be returned to its original place because of the `apiRef` not having the API methods on the first `useEffect` run. I've mentuned this {HERE](https://github.com/mui-org/material-ui-x/pull/930#issuecomment-768163834)

Reply:
> We can move the NoSsr into the AutoSizer

I don't understand. AutoSizer already disable server-side rendering.

Reply:
Yes but as @DanailH said, the interest of noSSr for us is that it renders after running the first round of effect. So we use the first round to build the apiRef for now, until we finish #939, then we would be able to put it in our own Autosizer

Reply:
@dtassone I still don't understand, why does it concern the AutoSizer?

Reply:
'it' is NoSsr.